Client-app unit tests/UserClientTests.cs

```diff
@@ -0,0 +1,112 @@
+using System.Collections.Generic;
+using System.Net;
+using System.Net.Http;
+using Client.Clients;
+using Client.Interfaces;
+using Newtonsoft.Json;
+using Xunit;
+
+namespace Client_app
+{
+    class HttpClientTester : IHttpService
+    {
+        public HttpResponseMessage Send(HttpRequestMessage message)
+        {
+            throw new System.NotImplementedException();
+        }
+
+        public HttpResponseMessage Get(string uri)
+        {
+            throw new System.NotImplementedException();
+        }
+
+        public HttpResponseMessage Post(string uri, HttpContent content)
+        {
+            string output = content.ReadAsStringAsync().Result;
+            // Parsed input
+            Dictionary<string, string> dictionary =
+                JsonConvert.DeserializeObject<Dictionary<string, string>>(output);
+
+            // Already existing users in the system
+            List<string> users = new List<string> {"username", "GenericUsername", "TotallyNotHacker"};
+
+            // Username and password, for the user "username"
+            string username = "username";
+            string hashedPassword = "$2a$11$i7v7o31a25APe3LUJ353F.OO/qy7ujlq.9SK3osK1.WSJReOTa.JS";
+
+
+            HttpResponseMessage responseMessage = new HttpResponseMessage();
+
+            // Endpoints
+            if (uri.EndsWith("login"))
+            {
+                if (username != dictionary["username"] ||
+                    !BCrypt.Net.BCrypt.Verify(dictionary["password"], hashedPassword))
+                    return new HttpResponseMessage(HttpStatusCode.BadRequest);
```

Comment:
OpenAPI specification siger ikke at vi har en BadRequest, i stedet bruger vi 404 Not Found